src/components/workflowEditor/nodes/CDNode.tsx

```diff
@@ -281,83 +282,84 @@ export class CDNode extends Component<CDNodeProps, CDNodeState> {
         const selectedNodeKey = `${this.props.selectedNode?.nodeType}-${this.props.selectedNode?.id}`
         const currentNodeKey = `${WorkflowNodeType.CD}-${this.props.id.substring(4)}`
         return (
-            <>
-                <Link to={this.props.to} onClick={this.onClickNodeCard} className="dc__no-decor">
-                    <div
-                        data-testid={`workflow-editor-cd-node-${this.props.environmentName}`}
-                        className={`workflow-node cursor ${this.props.deploymentAppDeleteRequest ? 'pl-0' : 'pl-16'}`}
-                    >
-                        {this.props.deploymentAppDeleteRequest ? (
-                            <div className="workflow-node__trigger-type-delete workflow-node__trigger-type--create-delete bcr-5 m-0 dc__position-abs fs-10 dc__uppercase dc__top-radius-8 dc__text-center" />
-                        ) : (
-                            <div className="workflow-node__trigger-type workflow-node__trigger-type--create">
-                                {this.props.triggerType}
-                            </div>
-                        )}
-                        <div className="workflow-node__title h-100 workflow-node__title--no-margin flex">
-                            <div className="workflow-node__full-width-minus-Icon p-12">
-                                <span className="workflow-node__text-light">
-                                    {this.props.deploymentAppDeleteRequest ? (
-                                        <div className="cr-5">
-                                            Deleting
-                                            <span className="dc__loading-dots" />
-                                        </div>
-                                    ) : (
-                                        this.props.title
-                                    )}
-                                </span>
-                                {envDescriptionTippy(this.props.environmentName, this.props.description)}
-                            </div>
-
-                            {/* TODO: Look into these css later */}
-                            <div
-                                className={`workflow-node__icon-common pt-12 pb-12 mr-12 ${
-                                    this.props.isVirtualEnvironment
-                                        ? 'workflow-node__CD-rocket-icon'
-                                        : 'workflow-node__CD-icon dc__flip'
-                                }`}
-                            />
+            <ConditionalWrap
+                condition={!this.props.isReadonlyView && !!this.props.to}
+                wrap={(children) => <Link to={this.props.to} onClick={this.onClickNodeCard} className="dc__no-decor">{children}</Link>}
```

Comment:
This will break in eslint with following error:
Do not define components during render. React will see a new component type on every render and destroy the entire subtree’s DOM nodes and state (https://reactjs.org/docs/reconciliation.html#elements-of-different-types). Instead, move this component definition out of the parent component “CDNode” and pass data as props. If you want to allow component creation in props, set allowAsProps option to true.